modules/plugins/ui/edgy-nvim/config.nix

```diff
@@ -0,0 +1,29 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  inherit (lib.modules) mkIf;
+  inherit (lib.strings) optionalString;
+  inherit (lib.nvim.dag) entryBefore;
+  inherit (lib.nvim.lua) toLuaObject;
+
+  cfg = config.vim.ui.edgy-nvim;
+in {
+  config = mkIf cfg.enable {
+    vim = {
+      startPlugins = ["edgy-nvim"];
+      pluginRC.edgy-nvim = entryBefore ["basic"] ''
+        require('edgy').setup(${toLuaObject cfg.setupOpts})
+
+        ${optionalString cfg.setRecommendedNeovimOpts ''
+          -- views can only be fully collapsed with the global statusline
+          vim.opt.laststatus = 3
+          -- Default splitting will cause your main splits to jump when opening an edgebar.
+          -- To prevent this, set `splitkeep` to either `screen` or `topline`.
+          vim.opt.splitkeep = "screen"
+        ''}
```

Comment:
Logically, it makes to put the option stuff before setup, so I'd do that.

Also, use `vim.o` instead of `vim.opt`. This is just being picky, but _technically_ `vim.o` is better performance wise, since `vim.opt` is a wrapper that does some extra stuff.